src/plugins/gpgme/testmod_gpgme.c

```diff
@@ -0,0 +1,286 @@
+/**
+ * @file
+ *
+ * @brief test suite for the gpgme plugin.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ * @author Peter Nirschl
+ *
+ */
+
+#include <gpgme.h>
+#include <kdbinternal.h>
+#include <locale.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tests_internal.h>
+#include <tests_plugin.h>
+#include <unistd.h>
+
+#include "gpgme.h"
+
+// GPG private key for importing and testing
+#include "test_key.h"
+
+#define TEST_KEY_ID "DDEBEF9EE2DC931701338212DAF635B17F230E8D"
+#define GPGME_PLUGIN_NAME "gpgme"
+#define KEYNAME_UNCHANGED "user/crypto/test/nochange"
+#define KEYNAME_NULL "user/crypto/test/mynull"
+#define KEYNAME_STRING "user/crypto/test/mystring"
+#define KEYNAME_BIN "user/crypto/test/mybin"
+
+static const char strVal[] = "gpgme test value";
+static const kdb_octet_t binVal[] = { 0xCA, 0xFE, 0xBA, 0xBE };
+
+static inline ssize_t MIN (ssize_t a, ssize_t b)
+{
+	return (a < b) ? a : b;
+}
+
+static void init_gpgme (void)
+{
+	gpgme_error_t err;
+
+	gpgme_check_version (NULL);
+	gpgme_set_locale (NULL, LC_CTYPE, setlocale (LC_CTYPE, NULL));
```

Comment:
Is this needed?

Reply:
I am not sure. It is explicitly recommended in the `gpgme` manual.